File: java-pr/49/java/performance/NumberProcessor16.java
// {fact rule=performance@v1.0 defects=1}
import java.util.*;
import java.util.stream.Collectors;
import java.io.*;

public class NumberProcessor16 {
    private Database database = new Database();
    
    public List<Integer> processNumbers(List<Integer> numbers) {
        List<Integer> result = new ArrayList<>();
        for (Integer num : numbers) {
            if (num % 2 == 0) {
                result.add(num * 2);
            }
        }
        return result;
    }
    
    private static class Database {
        public User findUserById(int id) { return new User(); }
        public List<User> findAllUsers() { return new ArrayList<>(); }
    }
    
    private static class User {}
}
// {/fact}

File: js-pr/56/javascript/code-quality-performance/number_processor_16.js
// {fact rule=code-quality-performance@v1.0 defects=1}
function processNumbers(numbers) {
    let result = [];
    for (let num of numbers) {
        if (num % 2 === 0) {
            result.push(num * 2);
        }
    }
    return result;
}
// {/fact}